Add missing configuration of the end-of-message detection register

Add setting of the "Timeout EOM Enable" bit to the hardware setup code.

--- a/drivers/tty/serial/lantiq.c
+++ b/drivers/tty/serial/lantiq.c
@@ -62,6 +62,7 @@
 #define LTQ_ASC_CON		0x0010
 #define LTQ_ASC_BG		0x0050
 #define LTQ_ASC_IRNREN		0x00F4
+#define LTQ_ASC_EOMCON		0x0080
 
 #define ASC_IRNREN_TX		0x1
 #define ASC_IRNREN_RX		0x2
@@ -108,6 +109,7 @@
 #define ASCFSTAT_TXFFLMASK	0x3F00
 #define ASCFSTAT_TXFREEMASK	0x3F000000
 #define ASCFSTAT_TXFREEOFF	24
+#define ASCEOMCON_TOEN		0x01000000
 
 static void lqasc_tx_chars(struct uart_port *port);
 static struct ltq_uart_port *lqasc_port[MAXPORTS];
@@ -328,6 +330,8 @@ lqasc_startup(struct uart_port *port)
 		((RXFIFO_FL << ASCRXFCON_RXFITLOFF) & ASCRXFCON_RXFITLMASK)
 		| ASCRXFCON_RXFEN | ASCRXFCON_RXFFLU,
 		port->membase + LTQ_ASC_RXFCON);
+	ltq_w32_mask(0, ASCEOMCON_TOEN,
+		port->membase + LTQ_ASC_EOMCON);
 	/* make sure other settings are written to hardware before
 	 * setting enable bits
 	 */
